source/GameInterface/Services/Settlements/Patches/LastAttackerPartySettlementPatch.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Policies;
+using GameInterface.Services.Settlements.Messages;
+using HarmonyLib;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Settlements;
+
+namespace GameInterface.Services.Settlements.Patches;
+
+/// <summary>
+/// Sync LastAttackerParty
+/// </summary>
+[HarmonyPatch(typeof(Settlement))]
+public class LastAttackerPartySettlementPatch
+{
+
+    [HarmonyPatch(nameof(Settlement.LastAttackerParty), MethodType.Setter)]
+    [HarmonyPrefix]
+    private static bool LastAttackerPartyPrefix(ref Settlement __instance, ref MobileParty value)
+    {
+        if (AllowedThread.IsThisThreadAllowed()) return true;
+        if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+        if (ModInformation.IsClient) return false;
```

Comment:
Change this to match up to line 42 of https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/GameInterface/Services/Armies/Patches/ArmyLifetimePatches.cs#L42